src/rgw/driver/rados/rgw_notify.cc

```diff
@@ -1236,33 +1249,43 @@ int publish_commit(rgw::sal::Object* obj,
           return ret;
         }
         // now try to make a bigger one
-	buffer::list obl;
-        int rval;
-        cls_2pc_queue_reserve(op, bl.length(), 1, &obl, &rval);
-        ret = rgw_rados_operate(
-	  dpp, res.store->getRados()->get_notif_pool_ctx(),
-          queue_name, &op, res.yield, librados::OPERATION_RETURNVEC);
+	// buffer::list obl;
+  //       int rval;
+  //       cls_2pc_queue_reserve(op, bl.length(), 1, &obl, &rval);
+  //       ret = rgw_rados_operate(
+	//   dpp, res.store->getRados()->get_notif_pool_ctx(),
+  //         queue_name, &op, res.yield, librados::OPERATION_RETURNVEC);
+        ret = rgw::redisqueue::reserve(conn, queue_name, bl.length(), res.yield);
         if (ret < 0) {
           ldpp_dout(dpp, 1) << "ERROR: failed to reserve extra space on queue: "
 			    << queue_name
 			    << ". error: " << ret << dendl;
           return (ret == -ENOSPC) ? -ERR_RATE_LIMITED : ret;
         }
-        ret = cls_2pc_queue_reserve_result(obl, topic.res_id);
-        if (ret < 0) {
-          ldpp_dout(dpp, 1) << "ERROR: failed to parse reservation id for "
-	    "extra space. error: " << ret << dendl;
-          return ret;
-        }
+      //   ret = cls_2pc_queue_reserve_result(obl, topic.res_id);
+      //   if (ret < 0) {
+      //     ldpp_dout(dpp, 1) << "ERROR: failed to parse reservation id for "
+	    // "extra space. error: " << ret << dendl;
+      //     return ret;
+      //   }
       }
-      std::vector<buffer::list> bl_data_vec{std::move(bl)};
+      // std::vector<buffer::list> bl_data_vec{std::move(bl)};
       librados::ObjectWriteOperation op;
-      cls_2pc_queue_commit(op, bl_data_vec, topic.res_id);
+
+      // FIXME: Is this commiting multiple messages over a single reservation?
```

Comment:
Hmm, I got confused because of the data type, `std::vector<buffer::list>`.  
I do not understand both parts of it. Why a vector and why is the message a list type in the buffer? The later could be how things are written in ceph. 

How do I get a raw string out of this? `decode`?